### PR TITLE
Setup ci

### PR DESCRIPTION
**Pull request by alex-jw-brooks**
This PR is essentially a copy + paste + find replace of @evaline-ju's caikit-nlp [PR](https://github.com/caikit/caikit-nlp/pull/2) - Thanks Evaline!

- Adds format script & formatter profiles to ensure they play together okay, runs the formatter
- Move towards pyproject.toml
- Enables GitHub workflow for building the library, linting etc
- Adds caikit contribution guidelines
- Adds Github templates

Note that currently the listing score is a bit low, so the fail threshold is 6; this will be brought up to 10 in a future PR

### CODE REVIEW

File: .isort.cfg [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[settings]
+profile=black
+from_first=true
+import_heading_future=Future
+import_heading_stdlib=Standard
+import_heading_thirdparty=Third Party
+import_heading_firstparty=First Party
+import_heading_localfolder=Local
+known_firstparty=alog,aconfig,caikit,caikit_computer_vision,import_tracker
```
Comment (gabe-l-hart) [inline .isort.cfg:9]: NIT: This looks like `caikit_computer_vision` is listed in both `firstparty` and `localfolder`

**Review by gabe-l-hart — APPROVED** [review]
Looks great Alex!

File: .isort.cfg (hunk 1 above)
Comment (alex-jw-brooks) [reply .isort.cfg]: Good catch, thanks!

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+[testenv:fmt]
+description = format with pre-commit
+deps = pre-commit>=3.0.4,<4.0
+commands = ./scripts/fmt.sh
+allowlist_externals = ./scripts/fmt.sh
+skip_install = True # Skip package install since fmt doesn't need to execute code, for ⚡⚡⚡
+
+[testenv:lint]
+description = lint with pylint
+deps = pylint>=2.16.2,<3.0
```
[the remark this answers is not in the document]
Comment (alex-jw-brooks) [reply tox.ini:33]: Good idea. Thanks!

**Pull request merged**